internal/services/apimanagement/api_management_api_data_source.go

```diff
@@ -129,50 +129,48 @@ func dataSourceApiManagementApiRead(d *pluginsdk.ResourceData, meta interface{})
 	defer cancel()
 
 	revision := d.Get("revision").(string)
-	id := parse.NewApiID(subscriptionId, d.Get("resource_group_name").(string), d.Get("api_management_name").(string), d.Get("name").(string))
+	apiId := fmt.Sprintf("%s;rev=%s", d.Get("name").(string), revision)
 
-	apiId := fmt.Sprintf("%s;rev=%s", id.Name, revision)
-	resp, err := client.Get(ctx, id.ResourceGroup, id.ServiceName, apiId)
+	id := api.NewApiID(subscriptionId, d.Get("resource_group_name").(string), d.Get("api_management_name").(string), apiId)
+	resp, err := client.Get(ctx, id)
 	if err != nil {
-		if utils.ResponseWasNotFound(resp.Response) {
-			return fmt.Errorf("api %s Revision %q does not exist", id, revision)
+		if response.WasNotFound(resp.HttpResponse) {
+			return fmt.Errorf("%s does not exist", id)
 		}
-
-		return fmt.Errorf("retrieving API %s / Revision %q : %+v", id, revision, err)
+		return fmt.Errorf("retrieving %s: %+v", id, err)
 	}
 
 	d.SetId(id.ID())
 
 	d.Set("api_management_name", id.ServiceName)
-	d.Set("name", id.Name)
-	d.Set("resource_group_name", id.ResourceGroup)
-
-	if props := resp.APIContractProperties; props != nil {
-		d.Set("description", props.Description)
-		d.Set("display_name", props.DisplayName)
-		d.Set("is_current", props.IsCurrent)
-		d.Set("is_online", props.IsOnline)
-		d.Set("path", props.Path)
-		d.Set("revision", props.APIRevision)
-		d.Set("service_url", props.ServiceURL)
-		d.Set("soap_pass_through", string(props.APIType) == string(apimanagement.SoapAPITypeSoapPassThrough))
-		d.Set("subscription_required", props.SubscriptionRequired)
-		d.Set("version", props.APIVersion)
-		d.Set("version_set_id", props.APIVersionSetID)
-
-		if err := d.Set("protocols", flattenApiManagementApiDataSourceProtocols(props.Protocols)); err != nil {
+	name := getApiName(id.ApiId)
+	d.Set("name", name)
+	d.Set("resource_group_name", id.ResourceGroupName)
+
+	if model := resp.Model; model != nil && model.Properties != nil {
+		d.Set("description", pointer.From(model.Properties.Description))
```

Comment:
For readability we assign `Properties` to a variable as well 
```suggestion
	if model := resp.Model; model != nil {
		if props := model.Properties; props != nil {
			d.Set("description", pointer.From(props.Description))
			...
		}
		
```

Reply:
Fixed.